cpp/src/arrow/type.h

```diff
@@ -67,9 +67,12 @@ struct Type {
     // Variable-length bytes (no guarantee of UTF8-ness)
     BINARY,
 
-    // By default, int32 days since the UNIX epoch
+    // int64_t milliseconds since the UNIX epoch
     DATE,
 
+    // int32_t days since the UNIX epoch
+    DATE32,
```

Comment:
Is the ordering of the types important yet, i.e. do we match the enums anywhere?